### ISSUE TITLE
PIVX core wallet does not update balance

### ISSUE THREAD

**Issue by SiggeBas** [PIVX-Project/PIVX#2371]
#### If so, please list the steps to reproduce below:
It happened twice so far, not sure when the count of incoming transactions stops, pretty sure that it doesn't stop directly after restart.

### Expected behavior
Incoming transactions and staking rewards should increase the total balance count,

### Actual behavior
Incoming transactions and staking rewards do not change the total amount of PIV. I need to close and restart the wallet to update.

### What version of PIVX Core are you using?
v5.1.0

### Machine specs:
- OS: Win10
- CPU: Intel i5 7200U
- RAM: 4
- Disk size: 1TB
- Disk Type (HD/SDD): SDD

### Any extra information that might be useful in the debugging process.
This is normally the contents of a `debug.log`, `db.log` or `config.log` file. Raw text or a link to a pastebin type site are preferred.

**Comment by SiggeBas** [comment]
I can now confirm that the balance does not update at all, only after restart of the wallet

**Comment by SiggeBas** [comment]
I today realized that the wallet will not completely close when I press X to close it. A process is still active, which I need to kill manually in order to restart the wallet.

**Comment by furszy** [comment]
Hello @SiggeBas, you are referring to two different topics.
About the second one, do you have the "Minimize on close" option enabled?

**Comment by jakimanboy** [comment]
I can confirm that "Not updating balance" also happened on my staking wallet few days back after upgrading to 5.1.0, and I had to restart the wallet to get it to show the correct balance. Windows 7 x64. Was going to open an issue if it happened again, but no issues since so far. 

At the time of the issue, both the Available balance and the Immature balance on the Home screen seemed like it was stuck even though the Transactions history was updating fine. So the wallet seemed to be still working / staking fine, but just the top balance display portion was not getting updated. 

P.S. I have another 5.1.0 upgraded wallet running on the same PC as a MN controller and that hasn't had any issues yet with updating its balance. So I didn't think much of it, but then I saw this opened github issue hence just commenting my similar experience.
